### PR TITLE
add reader to get the simple_oauth_header object

### PR DESCRIPTION
**Pull request by rivernate**

### CODE REVIEW

File: spec/ims/lti/models/messages/message_spec.rb [hunk 1]
```diff
@@ -199,5 +199,34 @@ class CustomMesage < Message
...
+      it 'returns the last simple_oauth_header_used' do
+        subject.launch_url = 'http://www.example.com'
+        params = subject.signed_post_params('secret')
+        message = described_class.new(params)
+        message.launch_url = 'http://www.example.com'
+        message.valid_signature?('secret')
+        expect(message.simple_oauth_header).to be_instance_of SimpleOAuth::Header
+      end
+
+      it 'returns nil if there has not been on used yet' do
```
Comment (abutterf) [inline spec/ims/lti/models/messages/message_spec.rb]: s/on/one

File: lib/ims/lti/models/messages/message.rb [hunk 2]
```diff
@@ -1,6 +1,8 @@
 module IMS::LTI::Models::Messages
   class Message
 
+    attr_reader :simple_oauth_header
```
Comment (abutterf) [inline lib/ims/lti/models/messages/message.rb]: Just so I understand why do we need to make this readable now?

File: spec/ims/lti/models/messages/message_spec.rb [hunk 3]
```diff
@@ -199,5 +199,34 @@ class CustomMesage < Message
 
     end
 
+    describe 'simple_oauth_header' do
+      it 'returns the last simple_oauth_header_used' do
```
Comment (abutterf) [inline spec/ims/lti/models/messages/message_spec.rb]: s/simple_oauth_header_used/simple_oauth_header used

File: lib/ims/lti/models/messages/message.rb (hunk 2 above)
Comment (rivernate) [inline lib/ims/lti/models/messages/message.rb]: It's too make it easier to debug OAuth signing issues. The current case is to compare base strings between TC and TP. I have been monkey patching, but decided to do something better.

**Pull request merged**